AndroidAnnotations/androidannotations-rest-spring/rest-spring/src/main/java/org/androidannotations/rest/spring/helper/RestAnnotationHelper.java

```diff
@@ -295,6 +301,29 @@ public JExpression getResponseClass(Element element, RestHolder holder) {
 		return responseClassExpr;
 	}
 
+	private boolean checkIfParameterizedTypeReferenceShouldBeUsed(TypeMirror returnType) {
+		switch (returnType.getKind()) {
+		case DECLARED:
+			return !((DeclaredType) returnType).getTypeArguments().isEmpty();
+
+		case ARRAY:
+			ArrayType arrayType = (ArrayType) returnType;
+			TypeMirror componentType = arrayType.getComponentType();
+			if (componentType.getKind() == TypeKind.DECLARED) {
```

Comment:
You could have returned `checkIfParameterizedTypeReferenceShouldBeUsed(componentType);` in all `if-else` cases.